### PR TITLE
Fix form state issue in remote log publishing

### PR DESCRIPTION
**Pull request by savindi7**
### Purpose
Fix the form state issue causing the state or value change of one form affecting the other.

### Related Issues
- https://github.com/wso2/product-is/issues/18855

### Related PRs
- Related PR `#1` or (None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/server/pages/remote-logging-page.tsx [hunk 1]
```diff
@@ -35,26 +36,29 @@ type RemoteLoggingPageInterface = IdentifiableComponentInterface;
...
 
     const {
         data: remoteLogPublishingConfigs,
         isLoading: isRemoteLogPublishingConfigsLoading,
-        error: remoteLogPublishingConfigRetrievalError
+        error: remoteLogPublishingConfigRetrievalError,
+        mutate: mutateRemoteLoggingRequest
     } = useRemoteLogPublishingConfigs();
 
+    const [ activeTab, setActiveTab ] = useState<TabProps["activeIndex"]>(0);
```
Comment (brionmario) [inline apps/console/src/features/server/pages/remote-logging-page.tsx:49]: Shall we use `tab-id` s instead.

```jsx
{
    "data-tabid": "audit-logs",
    menuItem: "Audit Logs",
    render: () => (
        <RemoteLoggingConfigForm
            logType={ LogType.AUDIT }
            logConfig={ remoteLogPublishingConfigs?.find(
                (config: RemoteLogPublishingConfigurationInterface) => config.logType === LogType.AUDIT
            ) }
        />    
    )
},
```

Comment (savindi7) [reply apps/console/src/features/server/pages/remote-logging-page.tsx]: fixed with https://github.com/wso2/identity-apps/pull/5183/commits/acfffd49ebcb00931594a3f8697d8a130e05f13e

File: apps/console/src/features/server/models/ui.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```
Comment (brionmario) [inline apps/console/src/features/server/models/ui.ts:2]:
```suggestion
 * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
```

**Review by DonOmalVindula — APPROVED** [review]

File: apps/console/src/features/server/components/remote-logging-config-form.tsx [hunk 3]
```diff
@@ -35,27 +38,31 @@ import React, { ReactElement, useState } from "react";
...
 import { Dispatch } from "redux";
-import { 
+import {
     restoreRemoteLogPublishingConfigurationByLogType,
     updateRemoteLogPublishingConfigurationByLogType
 } from "../api/server";
 import { LogType, RemoteLogPublishingConfigurationInterface } from "../models/server";
 
 interface RemoteLoggingConfigFormProps extends IdentifiableComponentInterface {
+    formId: "audit" | "carbon";
```
Comment (brionmario) [inline apps/console/src/features/server/components/remote-logging-config-form.tsx:48]: Lets remove this.

File: apps/console/src/features/server/models/ui.ts (hunk 2 above)
Comment (savindi7) [reply apps/console/src/features/server/models/ui.ts]: fixed with https://github.com/wso2/identity-apps/pull/5183/commits/3b7df545443c8f17b195fe6cf82890ed018ef01a

File: apps/console/src/features/server/components/remote-logging-config-form.tsx (hunk 3 above)
Comment (savindi7) [reply apps/console/src/features/server/components/remote-logging-config-form.tsx]: fixed with https://github.com/wso2/identity-apps/pull/5183/commits/bb554588d1bad946890ec7690bdf5e791370ee7e

File: apps/console/src/features/server/pages/remote-logging-page.tsx [hunk 4]
```diff
@@ -19,42 +19,47 @@
...
 
     const {
         data: remoteLogPublishingConfigs,
         isLoading: isRemoteLogPublishingConfigsLoading,
-        error: remoteLogPublishingConfigRetrievalError
+        error: remoteLogPublishingConfigRetrievalError,
+        mutate: mutateRemoteLoggingRequest
     } = useRemoteLogPublishingConfigs();
 
+    const [ activeTab, setActiveTab ] = useState<string>(RemoteLoggingTabIds.AUDIT);
```
[the remark this answers is not in the document]
Comment (savindi7) [reply apps/console/src/features/server/pages/remote-logging-page.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/5183/commits/bb554588d1bad946890ec7690bdf5e791370ee7e

**Review by DonOmalVindula — APPROVED** [review]

**Review by Achintha444 — APPROVED** [review]

**Review by NipuniBhagya — APPROVED** [review]

File: apps/console/src/features/server/models/ui.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Enum for Remote Logging tab IDs.
+ */
+export enum RemoteLoggingTabIds {
```
Comment (pavinduLakshan) [inline apps/console/src/features/server/models/ui.ts:22]:
```suggestion
export enum RemoteLoggingTab {
```

**Review by pavinduLakshan — APPROVED** [review]

File: apps/console/src/features/server/models/ui.ts (hunk 5 above)
Comment (savindi7) [reply apps/console/src/features/server/models/ui.ts:22]: we will fix this in a later commit.

**Pull request merged**